doc/api/vm.md

```diff
@@ -1573,6 +1573,84 @@ inside a `vm.Context`, functions passed to them will be added to global queues,
 which are shared by all contexts. Therefore, callbacks passed to those functions
 are not controllable through the timeout either.
 
+### Class: `NodeRealm`
+
+> Stability: 1 - Experimental. Use `--experimental-node-realm` CLI flag to
+> enable this feature.
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+A `NodeRealm` is effectively a Node.js environment that runs within the
+same thread. It similar to a [ShadowRealm][], but with a few main differences:
+
+* `NodeRealm` allows to load both commonjs and ESM modules.
+* Full interoperability between the host realm and the `NodeRealm` instance
+  is allowed
+* There is a deliberate `stop()` function.
+
+```mjs
+import { NodeRealm } from 'node:vm';
+const nodeRealm = new NodeRealm();
+const myAsyncFunction = nodeRealm.createImport(import.meta.url)('my-module');
```

Comment:
IIUC, the `import` function is asynchronous and resolves with a `ModuleNamespace`:

```suggestion
const { myAsyncFunction } = await nodeRealm.createImport(import.meta.url)('my-module');
```